### ISSUE TITLE
Angular localize support dynamic/runtime translations

### ISSUE THREAD

**Issue by Coconaut5** [angular/angular#56318]
### Which @angular/* package(s) are relevant/related to the feature request?

localize

### Description

Hello! To my understanding it isn't possible to translate dynamic values at runtime using angular/localize out of the box. Such as:

```
// component.ts
words = ['hello', 'world']

// component.html
@for (word of words; track word) {
<p i18n="@@{{word}}">
 {{word}}
</p>
}
```

I've also tried using the $localize in the ts file such as ```words=[$localize`hello`, $localize`world`]```

Neither of these is possible in angular/localize due to it being translated at compile time. Please correct me if im wrong. Is this something that would be possible to implement? Or am I mistaken on the capabilties?

Thank you! 💖

### Proposed solution

I don't know what would be a viable solution for Angular/Localize since the translations are generated at compile time? Im puzzled about how other applications handle translations with this package since my understanding is that this is not possible. I think a translate pipe would be very nice and easy to work with, but maybe it's not an easy task.

### Alternatives considered

I might very well be wrong, but in my scenario it seems like the only solution would be switching to a third-party library such as [transloco](https://jsverse.github.io/transloco/).

**Comment by chintankavathia** [comment]
This may help https://github.com/angular/angular/issues/46851

**Comment by Coconaut5** [comment]
@chintankavathia Yes thank you! I was trying something similarly earlier today but couldn't get it to work. But for some reason now i got it to work. However I noticed that this does actually change the value of the variable as well. Compared to when I've previously used ngx-translate and just used a pipe in the template. So that could lead to unintended behaviors if not aware of it. I'm not sure if that poses any issue though.
